workspaces/ui-v2/src/utils/pathToRegexpEscaped.ts

```diff
@@ -0,0 +1,41 @@
+import { pathToRegexp } from 'path-to-regexp';
+
+const specialCharactersRegex = /[.*+?^${}()|[\]\\]/g;
+
+// Special characters need to be escaped to be created as a regexp
+const replaceSpecialCharacters = (s: string) =>
+  s.replace(specialCharactersRegex, '\\$&');
+
+// The pathToRegexp lib doesn't treat `-` as the same path parameter
+const replaceDashWithUnderscoreInNamedParameter = (s: string) =>
+  s
+    .split('/')
+    .map((part) => (part.startsWith(':') ? part.replace(/-/g, '_') : part))
+    .join('/');
+
+// @gotcha this swallows query parameters - it's more appropriate that this function doesn't use query parameters
```

Comment:
This is a little odd, but the way that I imagine we have to handle query parameters is that it is treated separately from the path